packages/fx-core/src/component/generator/apiSpec/helper.ts

```diff
@@ -389,7 +389,11 @@ export async function generateFromApiSpec(
   projectType: ProjectType,
   outputFilePath: SpecParserOutputFilePath
 ): Promise<Result<SpecParserGenerateResult, FxError>> {
-  const operations = inputs[QuestionNames.ApiOperation] as string[];
+  const operations =
+    featureFlagManager.getBooleanValue(FeatureFlags.KiotaIntegration) &&
+    inputs[QuestionNames.ApiPluginManifestPath]
+      ? (await specParser.list()).APIs.filter((value) => value.isValid).map((value) => value.api)
+      : (inputs[QuestionNames.ApiOperation] as string[]);
```

Comment:
we do validation in line 397. Feels that it would make more sense if moving list() behind validate()

Reply:
I remember they are different checks. @SLdragon Do you know what's the difference with isValid and specParser.validate() ?